src/Nethermind/Nethermind.Specs/ChainSpecStyle/ChainSpecBasedSpecProvider.cs

```diff
@@ -249,46 +276,27 @@ protected virtual ReleaseSpec CreateReleaseSpec(ChainSpec chainSpec, long releas
                 item.ApplyToReleaseSpec(releaseSpec, releaseStartBlock, releaseStartTimestamp);
             }
 
-            SetMaxAndTargetBlobCount();
+            SetBlobScheduleParameters();
 
             return releaseSpec;
 
-            void SetMaxAndTargetBlobCount()
+            void SetBlobScheduleParameters()
             {
-                (IReleaseSpec? fork, IReleaseSpec? previousFork) = GetCurrentAndPreviousFork();
-                if (fork is null) return;
+                BlobScheduleSettings? blobSchedule = chainSpec.Parameters.BlobSchedule.OrderByDescending(bs => bs).FirstOrDefault(bs => bs.Timestamp <= releaseStartTimestamp);
```

Comment:
You still need to do `OrderByDescending`?
Also you do it for each release spec, can you do it once?
Also this is sorted, if it was array we could use binary search? (And you convert it to array somewhere later)

Reply:
Yep, the set stores it in ascending (expected by devs) order, but different order is required here.
I added check for 4844, before it it's not relevant. It decreases iterations from N*(N-1)/2 to 5*(5-1)/2 = 10 (N is all fork number, 5 is max supposed bpo fork count)